deps/rabbitmq_auth_backend_oauth2/src/rabbit_auth_backend_oauth2.erl

```diff
@@ -300,7 +300,7 @@ username_from(ClientProvidedUsername, DecodedToken) ->
 tags_from(DecodedToken) ->
     Scopes    = maps:get(<<"scope">>, DecodedToken, []),
     TagScopes = matching_scopes_without_prefix(Scopes, <<"tag:">>),
-    lists:usort(lists:map(fun rabbit_data_coercion:to_atom/1, TagScopes)).
+    lists:usort(lists:map(fun rabbit_data_coercion:to_existing_atom/1, TagScopes)).
```

Comment:
I'm not familiar with this plugin, so pardon the question if it's obvious... how do these tags get created before this function is called?

Reply:
They are not created but `to_existing_atom/1` should avoid allocating new atom table entries. They are just stored as a list otherwise.

Reply:
Hi,

The Tag comes from the JSON token which was created by an "Owner" supplied Authentication Service, like UAA in CloudFoundry. This value is controlled by the owner of that service and not by the User using the token, so it's all good even with `to_atom`, as we expect users to generate a valid. It is just an extra check to make sure the atom exists.